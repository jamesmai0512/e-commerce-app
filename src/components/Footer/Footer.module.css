.description {
  display: flex;
  flex: 1;
  justify-content: space-between;
  padding-right: 70px;
  padding-bottom: 40px;
}

.mail_content {
  padding-left: 30px;
  flex: 1;
}

.footer_content {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #4e4d93;
  text-align: start;
}

.content {
  padding: 58px 83px 14px 83px;
  display: flex;
  flex-direction: column;
}

.footer {
  background-color: #2a254b;
  color: white;
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  text-align: center;
  height: 380px;
  font-family: 'Satoshi Variable', sans-serif;
}

.title {
  font-size: 16px;
  line-height: 20px;
  padding-bottom: 12px;
}

.title_item {
  font-size: 14px;
  line-height: 19px;
  padding-bottom: 12px;
}

.copyright {
  font-size: 14px;
}

.title_main {
  padding-bottom: 16px;
}

.bottom {
  padding-top: 25px;
  display: flex;
  justify-content: space-between;
}

.list_item {
  display: flex;
  flex-direction: column;
  align-items: start;
}

.sponsor {
  display: flex;
  gap: 24px;
}
