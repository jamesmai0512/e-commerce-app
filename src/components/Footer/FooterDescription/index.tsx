import { CATEGORIES, MENU, OURCOMPANY } from '../../../constants/common'

// Styles
import styles from '../Footer.module.css'

const FooterDescription = () => {
  const { description, title, title_item, list_item } = styles
  return (
    <div className={description}>
      <div className={list_item}>
        <h2 className={title}>Menu</h2>
        {MENU.map((i) => (
          <p className={title_item} key={i}>
            {i}
          </p>
        ))}
      </div>
      <div className={list_item}>
        <h2 className={title}>Categories</h2>
        {CATEGORIES.map((i) => (
          <p className={title_item} key={i}>
            {i}
          </p>
        ))}
      </div>
      <div className={list_item}>
        <h2 className={title}>Our company</h2>
        {OURCOMPANY.map((i) => (
          <p className={title_item} key={i}>
            {i}
          </p>
        ))}
      </div>
    </div>
  )
}

export default FooterDescription
