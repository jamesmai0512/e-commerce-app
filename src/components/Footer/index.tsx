import { CATEGORIES, MENU, OURCOMPANY } from '../../constants/common'
import LinkedInIcon from '@mui/icons-material/LinkedIn'
import FacebookIcon from '@mui/icons-material/Facebook'
import InstagramIcon from '@mui/icons-material/Instagram'
import TwitterIcon from '@mui/icons-material/Twitter'
import PinterestIcon from '@mui/icons-material/Pinterest'

// Components
import Button from '../Button'
import Input from '../Input'

// Styles
import styles from './Footer.module.css'

const Footer = () => {
  const {
    footer,
    content,
    footer_content,
    description,
    title,
    title_item,
    mail_content,
    title_main,
    bottom,
    copyright,
    list_item,
    sponsor,
  } = styles
  return (
    <div className={footer}>
      <div className={content}>
        <div className={footer_content}>
          <div className={description}>
            <div className={list_item}>
              <h3 className={title}>Menu</h3>
              {MENU.map((i) => (
                <p className={title_item} key={i}>
                  {i}
                </p>
              ))}
            </div>
            <div className={list_item}>
              <h3 className={title}>Categories</h3>
              {CATEGORIES.map((i) => (
                <p className={title_item} key={i}>
                  {i}
                </p>
              ))}
            </div>
            <div className={list_item}>
              <h3 className={title}>Our company</h3>
              {OURCOMPANY.map((i) => (
                <p className={title_item} key={i}>
                  {i}
                </p>
              ))}
            </div>
          </div>
          <div className={mail_content}>
            <h2 className={title_main}>Join our mailing list</h2>
            <Input placeholder="your@email.com" />
            <Button background="light" text="Sign Up" />
          </div>
        </div>
        <div className={bottom}>
          <p className={copyright}>Copyright 2022 Avion LTD</p>
          <div className={sponsor}>
            <LinkedInIcon style={{ color: 'white' }} />
            <FacebookIcon style={{ color: 'white' }} />
            <InstagramIcon style={{ color: 'white' }} />
            <TwitterIcon style={{ color: 'white' }} />
            <PinterestIcon style={{ color: 'white' }} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default Footer
