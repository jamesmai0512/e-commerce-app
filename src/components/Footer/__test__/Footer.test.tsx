import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Footer from '../'

describe('Footer component', () => {
  it('Should render snapshot correctly', () => {
    const component = render(<Footer />)

    expect(component).toMatchSnapshot()
  })

  it('Should render an Input component', () => {
    render(<Footer />)
    const input = screen.getByPlaceholderText('your@email.com')
    expect(input).toBeInTheDocument()
  })

  it('Should contain the correct company name', () => {
    render(<Footer />)
    const companyName = screen.getByText('Copyright 2022 Avion LTD')
    expect(companyName).toBeInTheDocument()
  })
})
