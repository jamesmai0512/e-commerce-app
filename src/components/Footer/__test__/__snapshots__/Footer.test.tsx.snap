// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer"
      >
        <div
          class="content"
        >
          <div
            class="footer_content"
          >
            <div
              class="description"
            >
              <div
                class="list_item"
              >
                <h3
                  class="title"
                >
                  Menu
                </h3>
                <p
                  class="title_item"
                >
                  New arrivals
                </p>
                <p
                  class="title_item"
                >
                  Best sellers
                </p>
                <p
                  class="title_item"
                >
                  Recently viewed
                </p>
                <p
                  class="title_item"
                >
                  Popular this week
                </p>
                <p
                  class="title_item"
                >
                  All products
                </p>
              </div>
              <div
                class="list_item"
              >
                <h3
                  class="title"
                >
                  Categories
                </h3>
                <p
                  class="title_item"
                >
                  Crockery
                </p>
                <p
                  class="title_item"
                >
                  Furniture
                </p>
                <p
                  class="title_item"
                >
                  Homeware
                </p>
                <p
                  class="title_item"
                >
                  Plant pots
                </p>
                <p
                  class="title_item"
                >
                  Chairs
                </p>
                <p
                  class="title_item"
                >
                  Crockerare
                </p>
              </div>
              <div
                class="list_item"
              >
                <h3
                  class="title"
                >
                  Our company
                </h3>
                <p
                  class="title_item"
                >
                  About us
                </p>
                <p
                  class="title_item"
                >
                  Vacancies
                </p>
                <p
                  class="title_item"
                >
                  Contact us
                </p>
                <p
                  class="title_item"
                >
                  Privacy
                </p>
                <p
                  class="title_item"
                >
                  Returns policy
                </p>
              </div>
            </div>
            <div
              class="mail_content"
            >
              <h2
                class="title_main"
              >
                Join our mailing list
              </h2>
              <input
                class="input_email "
                data-testid="input-email"
                placeholder="your@email.com"
                type="email"
                value=""
              />
              <button
                class="button light"
                data-testid="button-component"
              >
                Sign Up
              </button>
            </div>
          </div>
          <div
            class="bottom"
          >
            <p
              class="copyright"
            >
              Copyright 2022 Avion LTD
            </p>
            <div
              class="sponsor"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LinkedInIcon"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FacebookIcon"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="InstagramIcon"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="TwitterIcon"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PinterestIcon"
                focusable="false"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.04 21.54c.96.29 1.93.46 2.96.46a10 10 0 0 0 10-10A10 10 0 0 0 12 2 10 10 0 0 0 2 12c0 4.25 2.67 7.9 6.44 9.34-.09-.78-.18-2.07 0-2.96l1.15-4.94s-.29-.58-.29-1.5c0-1.38.86-2.41 1.84-2.41.86 0 1.26.63 1.26 1.44 0 .86-.57 2.09-.86 3.27-.17.98.52 1.84 1.52 1.84 1.78 0 3.16-1.9 3.16-4.58 0-2.4-1.72-4.04-4.19-4.04-2.82 0-4.48 2.1-4.48 4.31 0 .86.28 1.73.74 2.3.09.06.09.14.06.29l-.29 1.09c0 .17-.11.23-.28.11-1.28-.56-2.02-2.38-2.02-3.85 0-3.16 2.24-6.03 6.56-6.03 3.44 0 6.12 2.47 6.12 5.75 0 3.44-2.13 6.2-5.18 6.2-.97 0-1.92-.52-2.26-1.13l-.67 2.37c-.23.86-.86 2.01-1.29 2.7v-.03z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer"
    >
      <div
        class="content"
      >
        <div
          class="footer_content"
        >
          <div
            class="description"
          >
            <div
              class="list_item"
            >
              <h3
                class="title"
              >
                Menu
              </h3>
              <p
                class="title_item"
              >
                New arrivals
              </p>
              <p
                class="title_item"
              >
                Best sellers
              </p>
              <p
                class="title_item"
              >
                Recently viewed
              </p>
              <p
                class="title_item"
              >
                Popular this week
              </p>
              <p
                class="title_item"
              >
                All products
              </p>
            </div>
            <div
              class="list_item"
            >
              <h3
                class="title"
              >
                Categories
              </h3>
              <p
                class="title_item"
              >
                Crockery
              </p>
              <p
                class="title_item"
              >
                Furniture
              </p>
              <p
                class="title_item"
              >
                Homeware
              </p>
              <p
                class="title_item"
              >
                Plant pots
              </p>
              <p
                class="title_item"
              >
                Chairs
              </p>
              <p
                class="title_item"
              >
                Crockerare
              </p>
            </div>
            <div
              class="list_item"
            >
              <h3
                class="title"
              >
                Our company
              </h3>
              <p
                class="title_item"
              >
                About us
              </p>
              <p
                class="title_item"
              >
                Vacancies
              </p>
              <p
                class="title_item"
              >
                Contact us
              </p>
              <p
                class="title_item"
              >
                Privacy
              </p>
              <p
                class="title_item"
              >
                Returns policy
              </p>
            </div>
          </div>
          <div
            class="mail_content"
          >
            <h2
              class="title_main"
            >
              Join our mailing list
            </h2>
            <input
              class="input_email "
              data-testid="input-email"
              placeholder="your@email.com"
              type="email"
              value=""
            />
            <button
              class="button light"
              data-testid="button-component"
            >
              Sign Up
            </button>
          </div>
        </div>
        <div
          class="bottom"
        >
          <p
            class="copyright"
          >
            Copyright 2022 Avion LTD
          </p>
          <div
            class="sponsor"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FacebookIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="InstagramIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TwitterIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PinterestIcon"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M9.04 21.54c.96.29 1.93.46 2.96.46a10 10 0 0 0 10-10A10 10 0 0 0 12 2 10 10 0 0 0 2 12c0 4.25 2.67 7.9 6.44 9.34-.09-.78-.18-2.07 0-2.96l1.15-4.94s-.29-.58-.29-1.5c0-1.38.86-2.41 1.84-2.41.86 0 1.26.63 1.26 1.44 0 .86-.57 2.09-.86 3.27-.17.98.52 1.84 1.52 1.84 1.78 0 3.16-1.9 3.16-4.58 0-2.4-1.72-4.04-4.19-4.04-2.82 0-4.48 2.1-4.48 4.31 0 .86.28 1.73.74 2.3.09.06.09.14.06.29l-.29 1.09c0 .17-.11.23-.28.11-1.28-.56-2.02-2.38-2.02-3.85 0-3.16 2.24-6.03 6.56-6.03 3.44 0 6.12 2.47 6.12 5.75 0 3.44-2.13 6.2-5.18 6.2-.97 0-1.92-.52-2.26-1.13l-.67 2.37c-.23.86-.86 2.01-1.29 2.7v-.03z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
